### PR TITLE
Add Animated Dock Component for Enhanced Navigation

### PR DESCRIPTION
**Pull request by R0X4R**
Hello!

### Description
I would like to propose the addition of an `Animated Dock` component to enhance the navigation experience in the project. This component provides a responsive and visually engaging navigation dock that works seamlessly on both desktop and mobile devices.

### Animation Preview
- Inspired By:

[Inspired Video](https://github.com/user-attachments/assets/7ab8afd8-ec17-4c69-85a4-8cbfa0fa3fb2)

Magnified Dock by [Build UI](https://buildui.com/recipes/magnified-dock)

- My Design:

[Preview Video](https://github.com/user-attachments/assets/d5014b25-d1d0-4722-85ba-9f24a3c5559e)

### Features of the AnimatedDock Component:
- **Large Dock:** Features icons that dynamically resize based on mouse proximity, creating an interactive experience.
Displays tooltips on hover, providing context for each navigation option.
- **Small Dock:** Includes a toggle button to expand or collapse the navigation options for mobile users.
Icons are presented in a vertical layout when expanded, ensuring ease of access on smaller screens.

### Implementation Details:
- The component is built using Framer Motion for smooth animations and transitions.
- It utilizes responsive design principles to adapt to different screen sizes.
- The code includes appropriate comments for clarity and maintainability.

### Benefits:
- The `Animated Dock` component will enhance user engagement through its interactive design.
- It improves navigation efficiency, making it easier for users to access various parts of the application.
- I believe this addition would significantly improve the overall user experience. I am happy to assist with any discussions or revisions regarding this feature.

Thank you for considering this contribution!

### CODE REVIEW

File: content/docs/container/animated-dock.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+title: Animated Dock
+description: A sleek dock-style navigation bar, inspired by macOS, that combines glassmorphic design with functionality. With smooth animations and responsive icons, it enhances navigation for a modern web application.
+author: Eshan Singh
```
Comment (hari) [inline content/docs/container/animated-dock.mdx:4]: it should be your twitter username @R0X4R

**Review by hari — APPROVED** [review]

File: content/docs/container/animated-dock.mdx (hunk 1 above)
Comment (R0X4R) [reply content/docs/container/animated-dock.mdx:4]: Okay let me add my twitter username

**Comment by hari** [comment]
@R0X4R looks like you are using `react-icons`, please use `lucide-react`

**Comment by R0X4R** [comment]
Okay let me change it to lucide-react

**Comment by R0X4R** [comment]
I have applied the changes

**Comment by R0X4R** [comment]
@hari  please check if the issues are fixed

**Review by hari — APPROVED** [review]

**Comment by R0X4R** [comment]
Thank you so much for approving my PR

**Pull request merged**